### PR TITLE
Automate creation of vertx artifacts list which are used in quarkus

### PR DESCRIPTION
**Pull request by SaumyaSingh1**
CC @RotBolt @harshmadhani 
This is a draft PR for "Automate the creation of vertx artifacts list which are used in quarkus".

### CODE REVIEW

**Comment by SaumyaSingh1** [comment]
Mainly findVertxArtifacts() will do the work :)

```
private List<String> findVertxArtifacts() {
        List<Path> allQuarkusJars = findAllByExtension("jar");
        List<Path> allVertxJarsInQuarkus = new ArrayList<>();

        for (Path jar : allQuarkusJars) {
            if (jar.toString().toLowerCase().contains("vertx")) {
                allVertxJarsInQuarkus.add(jar);
            }
        }

        Set<String> extensionsJson = extractExtensionsJsonArtifactIds();
        return allVertxJarsInQuarkus.stream()
            .filter(this::hasQuarkusExtensionMetadata)
            .map(this::extractArtifactId)
            .filter(extensionsJson::contains)
            .filter(ext -> !skipped.contains(ext))
            .collect(Collectors.toList());
    }
```

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
@SaumyaSingh1 It's just formatting diffs causing the CI test to be upset again. Could you please run:

```
mvn clean install -DskipTests=true
```

in your branch and rebase the changes to the commit to properly format your code please? Thank you! :smile:

**Comment by SaumyaSingh1** [comment]
> @SaumyaSingh1 It's just formatting diffs causing the CI test to be upset. Could you please run:
> 
> ```
> mvn clean install -DskipTests=true
> ```
> 
> in your branch and rebase the changes to the commit to properly format your code please? Thank you!

Yeah, I just ran the command and now it's properly formatted :)
Thank you @thescouser89

**Comment by SaumyaSingh1** [comment]
Hi @thescouser89 ,

These are the final changes and is locally tested, working as expected :)

File: pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/quarkus/VertxArtifactFinder.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.jboss.pnc.bacon.pig.impl.utils.FileUtils.mkTempDir;
+
+/**
+ * @author Saumya Singh, singhsaumyas150@gmail.com <br>
+ *         Date: 25/03/2021
+ */
+
```
Comment (thescouser89) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/quarkus/VertxArtifactFinder.java:32]: Hi @SaumyaSingh1 !

I know we're not consistent enough with this, but could you please add a description of the Add-on here that we could then also put in our documentation? Thank you! :)

Comment (thescouser89) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/quarkus/VertxArtifactFinder.java:32]: Otherwise LGTM! :rocket:

Comment (SaumyaSingh1) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/quarkus/VertxArtifactFinder.java:32]: Sure thing @thescouser89 :) 
I should have added a description earlier.

>we could then also put in our documentation

If you could direct me to the link/doc where I have to add details of this Add-on, I would love to do that :)

Comment (thescouser89) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/quarkus/VertxArtifactFinder.java:32]: Thank you!

The specific section is: https://project-ncl.github.io/bacon/guide/pig.html#add-ons

Which is from: https://github.com/project-ncl/bacon/blob/gh-pages/guide/pig.md#add-ons

It's the same repository, but in the `gh-pages` branch. So you can create your own branch from `gh-pages`, then do the changes to the `pig.md` file, and create a PR targetting the `gh-pages` branch.

As you can see, we're doing a really poor job at maintaining the add-on section. Thank you for your help on improving our documentation! :heart:

Comment (SaumyaSingh1) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/quarkus/VertxArtifactFinder.java:32]: Sure thing Dustin, I will make a PR for it :) 
I really liked the neat documentation of Bacon https://project-ncl.github.io/bacon/.

Comment (thescouser89) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/quarkus/VertxArtifactFinder.java:32]: That's all thanks to @rnc ! He's the magician behind it :)

**Pull request merged**